packages/server/src/workers/post-deploy-migration.ts

```diff
@@ -61,6 +65,17 @@ export const initPostDeployMigrationWorker: WorkerInitializer = (config) => {
   return { queue, worker, name: PostDeployMigrationQueueName };
 };
 
+export async function isClusterCompatible(migrationNumber: number): Promise<boolean> {
+  if (!enforceStrictMigrationVersionChecks()) {
+    return true;
+  }
+
+  const servers = await getRegisteredServers(true);
+  const entry = getPostDeployManifestEntry(migrationNumber);
+  const requiredVersion = entry.serverVersion;
+  return servers.every((server) => semver.gte(server.version, requiredVersion));
```

Comment:
For those following `main` rather than waiting for versioned releases like api.medplum.com, the expected behavior from this check is that the cluster will be considered incompatible until the official `entry.serverVersion` release, e.g. `4.3.0`, is deployed. The result will be the post-deploy migration repeatedly delayed in the interim.

While not ideal, that's shouldn't necessarily be problematic. We may want to come up with an alternative that isn't strictly based on numbered releases. Hmmm...